### PR TITLE
Enable load results manually from any available iteration

### LINKED ISSUE #8640: Not possible to load results manually from other iteration than "ensemble.iteration"
### What happened? (You can include a screenshot if it helps explain)

Currently no matter what value you set for `iteration` in the load results manually tool, it will only try to load from the iteration that is connected to the ensemble. If you have manually created a new ensemble, this is usually 0.
Reason is the following code which is used to load results from runpath, in `libres_facade.py`:
```
    def load_from_forward_model(
        self,
        ensemble: Ensemble,
        realisations: npt.NDArray[np.bool_],
        iteration: Optional[int] = None,
    ) -> int:
        if iteration is not None:
            warnings.warn(
                "The iteration argument has no effect, iteration is read from ensemble",
                DeprecationWarning,
                stacklevel=1,
            )
```

### What did you expect to happen?

Would expect the tool to load from the iteration specified.

### steps to reproduce

```Python
ert gui poly.ert
run an es_mda run
Create new experiment (and ensemble) in manage experiment
Try to load results manually into new ensemble, with iteration>0
Can see in the plotter that the result is from iteration 0
```

### Environment where bug has been observed

- [ ] python 3.8
- [ ] python 3.9
- [ ] python 3.10
- [X] python 3.11
- [ ] python 3.12
- [x] macosx
- [ ] rhel7
- [X] rhel8
- [X] local queue
- [ ] lsf queue
- [ ] slurm queue
- [ ] openPBS queue
- [ ] used flag `--enable-scheduler`

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #8640 

(Screenshot of new behavior in GUI if applicable)

Loading data
image

Checking plot results:
image

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Review by yngve-sk — COMMENTED** [review]
Sure we don't want a dropdown to enforce the selection of a valid iteration, rather than a text field (unless you want to be able to select multiple iterations etc?)

**Comment by DanSava** [comment]
I created the issue https://github.com/equinor/ert/issues/8819 to keep track of the dropdown conversion idea.

**Review by oyvindeide — COMMENTED** [review]
Think we need to rethink a bit how we do this, so everything is kept in sync.

File: src/ert/libres_facade.py [hunk 1]
```diff
@@ -128,12 +128,6 @@ def load_from_forward_model(
         realisations: npt.NDArray[np.bool_],
         iteration: Optional[int] = None,
     ) -> int:
-        if iteration is not None:
```
[the remark this answers is not in the document]
Comment (DanSava) [reply src/ert/libres_facade.py:131]: I might be missing something but as far as I can tell this will not change or modify the ensemble iteration number.

File: src/ert/run_arg.py [hunk 2]
```diff
@@ -29,8 +29,10 @@ def create_run_arguments(
     runpaths: Runpaths,
     active_realizations: Union[List[bool], npt.NDArray[np.bool_]],
     ensemble: Ensemble,
+    iteration: Optional[int] = None,
 ) -> List[RunArg]:
-    iteration = ensemble.iteration
+    if iteration is None:
+        iteration = ensemble.iteration
```
[the remark this answers is not in the document]
Comment (DanSava) [reply src/ert/run_arg.py:35]: This is an interesting question. Now that I think about it. It could be abused to create a `runpath` containing an iteration that does not match the `ensemble` iteration. I will try removing this change and working around it.

[the remark this answers is not in the document]
Comment (DanSava) [reply src/ert/run_arg.py:35]: Ok give it a try and if I remove this I will need to add another helper function to create a `List[RunArg]` that will be used to load from data from a forward model custom iteration. This new function could also be used to make a runpath where the ensemble iteration does not match the runpath iteration. So I am unsure how much this will help prevent future headache if somebody really wants to have an ensemble iteration out of sync with the runpath iteration. 

As far as I can tell this can also be achieved now if somebody in the code decides to update the ensemble iteration number directly. They will be out of sync with the runpath iteration and there is no hard check for this as far as I could find.

**Comment by DanSava** [comment]
Some tests are still needed for some of the new functionality but the implementation is ready for a review.

**Comment by yngve-sk** [comment]
Edge case: If invalid iteration is entered it still shows as successful
1. Create empty experiment -> ensemble (both named `test`) 
2. Load results manually into it, "select" iteration 6, and I get this: 

Screenshot 2024-10-11 at 10 11 02

**Comment by yngve-sk** [comment]
It seems to still load iteration 0 for me whenever I select a different iteration
Screenshot 2024-10-11 at 10 17 04
Screenshot 2024-10-11 at 10 17 39
Screenshot 2024-10-11 at 10 17 57
Screenshot 2024-10-11 at 10 18 19

**Comment by oyvindeide** [comment]
Doing some testing, and looks good so far! It would be great if I also could see what `<ITER>` would be replaced by in the runpath (read only), adding a text field with for example:

```python
f"<IENS> will be replace by: {ensemble.iteration}"
```

File: src/ert/callbacks.py [hunk 3]
```diff
@@ -128,3 +134,46 @@ async def forward_model_ok(
         run_arg.ensemble_storage.unset_failure(run_arg.iens)
 
     return final_result
+
+
+async def load_run_path_realization(
```
Comment (oyvindeide) [inline src/ert/callbacks.py:139]: Why do we need a new function instead of using `forward_model_ok`? If because of `RunArg`, could we just use the new function everywhere?

File: src/ert/libres_facade.py [hunk 4]
```diff
@@ -122,56 +121,47 @@ def get_ensemble_size(self) -> int:
...
-        iteration: Optional[int] = None,
     ) -> int:
-        if iteration is not None:
-            warnings.warn(
-                "The iteration argument has no effect, iteration is read from ensemble",
-                DeprecationWarning,
-                stacklevel=1,
-            )
         t = time.perf_counter()
-        run_args = create_run_arguments(
```
Comment (oyvindeide) [inline src/ert/libres_facade.py:138]: Nice improvement that we dont need `RunArgs` anymore!

**Review by oyvindeide — COMMENTED** [review]
Looks like a good improvement to me, a few minor questions

File: src/ert/callbacks.py (hunk 3 above)
Comment (DanSava) [reply src/ert/callbacks.py:139]: Yes, but I did not want to mess with `forward_model_ok`

**Review by oyvindeide — APPROVED** [review]
LGTM!
